onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -508,8 +540,8 @@ static bool RemoveBackToBackCasts(Graph& graph, Node* parent,
         // No more children nodes exists, and the parent-cast output is not a graph output. Remove it!
         ORT_THROW_IF_ERROR(RemoveCastNodesChain(graph, {parent}, removed_nodes));
       }
-      if (new_consumers.size() > 0) {
-        std::vector<Node*> consumers = graph.GetMutableConsumerNodes(parent_input->Name());
+      if (!new_consumers.empty()) {
```

Comment:
Considering that the empty() is implemented using size() as in inlined_vector.h
```  // InlinedVector::empty()
  //
  // Returns whether the inlined vector contains no elements.

  bool empty() const noexcept { return !size(); }
```
Is it required to replace` container.size()>0` with` !contaner.empty() `?

Reply:
This is causing an additional function call overhead.

Reply:
See my response in other similar comment.